datanode/broker/sqlstore_broker.go

```diff
@@ -119,8 +130,16 @@ func (b *SQLStoreBroker) Receive(ctx context.Context) error {
 		if err != nil {
 			return fmt.Errorf("failed to receive events, unable to get chain id:%w", err)
 		}
-		lastCommittedBlockHeight := nextBlock.Height - 1
-		if b.onBlockCommitted(ctx, chainID, lastCommittedBlockHeight) {
+
+		if b.onBlockCommitted(ctx, chainID, b.lastBlock.Height) {
+			if b.receivedProtocolUpgradeEvent {
```

Comment:
Is this code ever called? It looks like snapshotService.OnBlockCommitted only ever returns false.. and even if that wasn't the case, don't we get equivalent behaviour by  just moving

```
		if b.receivedProtocolUpgradeEvent {
			b.protocolUpgradeService.SetProtocolUpgradeStarted()
			return nil
		}
``` 
above this block?